src/Components/Authentication/Desktop/ModalManager.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react"
+import { DesktopModal } from "./Components/DesktopModal"
+import { FormSwitcher } from "./FormSwitcher"
+import { handleSubmit } from "Components/Authentication/helpers"
+import { ModalType } from "Components/Authentication/Types"
+
+interface Props {
+  submitUrls: { [P in ModalType]: string }
```

Comment:
@l2succes curious, what does this signature do?

Reply:
@damassi  `ModalType` is an `enum` type that looks like the following. 
```typescript
enum ModalType {
  login = "login"
  signup = "signup"
}
```
`{ [P in ModalType]: string }` essentially tells typescript that the values in the enum (denoted as `P`) are the only accepted keys in the `submitUrls` object.